core/app/helpers/spree/base_helper.rb

```diff
@@ -57,8 +57,29 @@ def logo(image_path = nil, options = {})
       end
     end
 
+    def object
+      instance_variable_get('@' + controller_name.singularize)
+    end
+
+    def og_meta_data
+      og_meta = {}
+      if object.is_a? Spree::Product
+        image                             = default_image_for_product_or_variant(object)
+        og_meta['og:image']               = main_app.url_for(image.attachment) if image&.attachment
+
+        og_meta['og:url']                 = spree.url_for(object) if frontend_available? # url_for product needed
+        og_meta['og:type']                = object.class.name.demodulize.downcase
+        og_meta['og:title']               = object.name
+        og_meta['og:description']         = object.description
+
+        og_meta['product:price:amount']   = object.master.default_price.amount
+        og_meta['product:price:currency'] = object.master.default_price.currency
```

Comment:
```suggestion
        if (price = object.price_in(current_currency)
          og_meta['product:price:amount']   = price.amount
          og_meta['product:price:currency'] = current_currency
        end
```

Reply:
@damianlegawiec followed your suggestion, can be reviewed once again now